code/starfield/starfield.cpp

```diff
@@ -242,24 +228,39 @@ void stars_load_debris_vclips(debris_vclip *vclips)
 	}
 }
 
-void stars_load_debris(int fullneb)
+void stars_load_debris(int fullneb, SCP_string custom_name)
 {
-	if (!Motion_debris_enabled) {
+	if (!Motion_debris_enabled || ((int)Motion_debris.size() == 0)) {
 		return;
 	}
 
-	// if we're in nebula mode
-	if ( fullneb && (stars_debris_loaded != 2) ) {
-		stars_release_debris_vclips(Debris_vclips);
-		stars_load_debris_vclips(Debris_vclips_nebula);
-		Debris_vclips = Debris_vclips_nebula;
-		stars_debris_loaded = 2;
-	} else if (stars_debris_loaded != 1) {
-		stars_release_debris_vclips(Debris_vclips);
-		stars_load_debris_vclips(Debris_vclips_normal);
-		Debris_vclips = Debris_vclips_normal;
-		stars_debris_loaded = 1;
+	SCP_string debris_name = "";
+	
+	if (!custom_name.empty()) {
+		debris_name = custom_name;
+	}
+
+	// if not using custom the load the default type for the mission
+	if (fullneb && debris_name.empty()) { // if we're in nebula mode
+		debris_name = "nebula";
+	} else if (debris_name.empty()) {
+		debris_name = "default";
 	}
+
+	int debris_position = get_motion_debris_by_name(debris_name);
```

Comment:
It's a bit pedantic but I do prefer the more explicit `index` in these cases instead of the more vague `position`.

Reply:
Done